apps/fluent-tester/src/FluentTester/FluentTester.android.tsx

```diff
@@ -0,0 +1,115 @@
+MessageQueue.spy(true);
+*/
+
+const DisplayIfVisible = ({ isVisible, children }) => (isVisible ? <View>{children}</View> : null);
+
+export interface FluentTesterProps {
+  initialTest?: string;
+  enabledTests: TestDescription[];
+}
+
+const Header: React.FunctionComponent<{}> = () => {
+  const theme = useTheme();
+
+  return (
+    <View style={{ ...fluentTesterStyles.header, flexDirection: 'column' }}>
+      <Text style={[fluentTesterStyles.testHeader]} variant="heroSemibold" color={theme.host.palette?.TextEmphasis} testID={BASE_TESTPAGE}>
+        ⚛ FluentUI Tests
+      </Text>
+      <ScrollView horizontal={true} showsHorizontalScrollIndicator={false}>
+        <ThemePickers />
+      </ScrollView>
+    </View>
+  );
+};
+
+const getThemedStyles = themedStyleSheet((t: Theme) => {
+  return {
+    root: {
+      backgroundColor: t.colors.background,
+      flexDirection: 'column',
+      height: '100%',
+      justifyContent: 'flex-start',
+      alignItems: 'stretch',
+      padding: 4,
+    },
+  };
+});
+
+export const FluentTester: React.FunctionComponent<FluentTesterProps> = (props: FluentTesterProps) => {
+  // sort tests alphabetically by name
+  const sortedTestComponents = props.enabledTests.sort((a, b) => a.name.localeCompare(b.name));
+
+  const [selectedTestIndex, setSelectedTestIndex] = React.useState(0);
+  const [onMainScreen, setOnMainScreen] = React.useState(true);
+
+  const TestComponent = sortedTestComponents[selectedTestIndex].component;
+
+  const themedStyles = getThemedStyles(useTheme());
+
+  const onBackPress = () => {
+    setOnMainScreen(true);
+    BackHandler.removeEventListener('hardwareBackPress', onBackPress);
+    return true;
+  };
+
+  return (
+    <DisplayIfVisible isVisible={true}>
+      <DisplayIfVisible isVisible={onMainScreen}>
+        <View style={themedStyles.root}>
+          <Header />
+          <Separator />
+          <View style={{ ...fluentTesterStyles.testRoot, marginTop: 10 }}>
+            <ScrollView style={fluentTesterStyles.testList} contentContainerStyle={fluentTesterStyles.testListContainerStyle}>
+              {sortedTestComponents.map((description, index) => {
+                return (
+                  <View key={index}>
+                    <Text
+                      key={index}
+                      onPress={() => {
+                        setOnMainScreen(false);
+                        setSelectedTestIndex(index);
+                        BackHandler.addEventListener('hardwareBackPress', onBackPress);
+                      }}
+                      style={mobileStyles.testListItems}
+                      testID={description.testPage}
+                    >
+                      {description.name}
+                    </Text>
+                    <Separator style={mobileStyles.testListSeparator} />
+                  </View>
+                );
+              })}
+            </ScrollView>
```

Comment:
I feels like this could be separated out into a separate component so that we don't have a block here. That also might help it be re-usable if it's split out